core/src/components/searchbar/searchbar.tsx

```diff
@@ -609,13 +609,60 @@ export class Searchbar implements ComponentInterface {
     return true;
   }
 
+  /**
+   * Get the icon to use for the clear icon.
+   * If an icon is set on the component, use that.
+   * Otherwise, use the icon set in the config.
+   * If no icon is set in the config, use the default icon.
+   *
+   * @returns {string} The icon to use for the clear icon.
+   */
+  get searchbarClearIcon(): string {
+    const theme = getIonTheme(this);
+    const icon = this.clearIcon;
+    const defaultIcon = theme === 'ios' ? closeCircle : closeSharp;
+
+    if (icon !== undefined) {
+      // Icon is set on the component.
+      return icon;
+    }
+
+    return config.get('searchbarClearIcon', defaultIcon);
+  }
+
+  /**
+   * Get the icon to use for the search icon.
+   * If an icon is set on the component, use that.
+   * Otherwise, use the icon set in the config.
+   * If no icon is set in the config, use the default icon.
+   *
+   * @returns {string} The icon to use for the search icon.
+   */
+  get searchbarSearchIcon(): string {
+    const theme = getIonTheme(this);
+    const icon = this.searchIcon;
+    const defaultIcon = theme === 'ios' ? searchOutline : searchSharp;
+
+    if (icon !== undefined) {
+      // Icon is set on the component.
+      return icon;
+    }
+
+    return config.get('searchbarSearchIcon', defaultIcon);
+  }
+
   render() {
-    const { cancelButtonText, autocapitalize } = this;
+    const { cancelButtonText, autocapitalize, searchbarClearIcon, searchbarSearchIcon } = this;
     const animated = this.animated && config.getBoolean('animated', true);
     const theme = getIonTheme(this);
-    const clearIcon = this.clearIcon || (theme === 'ios' ? closeCircle : closeSharp);
-    const searchIcon = this.searchIcon || (theme === 'ios' ? searchOutline : searchSharp);
     const shouldShowCancelButton = this.shouldShowCancelButton();
+    /**
+     * The back button icon will be the default if available when
+     * the searchbar cancel icon is not set.
+     * This provides a backword compatibility since the back button
+     * icon was the default cancel button icon before.
```

Comment:
```suggestion
     * The `backButtonIcon` config will be used as a fallback if the
     * `searchbarCancelIcon` config is not set. This ensures apps
     * have a way to keep the back button icon and searchbar cancel 
     * icon in sync.
```

I don't think we need to mention backwards compatibility here since this config just exists to keep the components in sync from a design perspective.